config/webpack.dev.config.js

```diff
@@ -17,6 +17,60 @@ dotenv.config({
   path: path.resolve(process.cwd(), '.env.development'),
 });
 
+/*
+This function reads in a 'module.config.js' file if it exists and uses its contents to define
+a set of webpack resolve.alias aliases for doing local development of application dependencies.
+It reads the package.json file of the dependency to determine if it has any peer dependencies, and
+then forces those peer dependencies to be resolved with the application's version.  Primarily, this
+is useful for making sure there's only one version of those dependencies loaded at once, which is a
+problem with both react and react-intl.
+
+The module.config.js file should have the form:
+
+{
+  localModules: [
+    { moduleName: 'nameOfPackage', dir: '../path/to/repo', dist: '/path/to/dist/in/repo' },
+    ... others...
+  ],
+}
+
+Some working examples, as of the time of this writing:
+
+{ moduleName: '@edx/paragon/scss', dir: '../paragon', dist: 'scss' }
+{ moduleName: '@edx/paragon', dir: '../paragon', dist: 'dist' }
+{ moduleName: '@edx/frontend-platform', dir: '../frontend-platform', dist: 'dist' }
+
+*/
+function getLocalAliases() {
+  const aliases = {};
+
+  try {
+    // eslint-disable-next-line import/no-dynamic-require, global-require
+    const { localModules } = require(path.resolve(process.cwd(), 'module.config.js'));
+
+    let allPeerDependencies = [];
+    const excludedPeerPackages = [];
+    localModules.forEach(({ moduleName, dir, dist = '' }) => {
+      // eslint-disable-next-line import/no-dynamic-require, global-require
+      const { peerDependencies, name } = require(path.resolve(process.cwd(), dir, 'package.json'));
+      allPeerDependencies = allPeerDependencies.concat(Object.keys(peerDependencies));
+      aliases[moduleName] = path.resolve(process.cwd(), dir, dist);
```

Comment:
why do you need `dist` here?

Reply:
if you remove this, can you remove the `dist` in the module config?

Reply:
Cause that's the path where you go to get the imports.  These webpack aliases are more literal than updating the dependency path in a package.json file, so they don't know how to use `main`, basically.

Reply:
Note that depending on the repo structure/how it builds, a particular dependency may choose for `dist` to be `''`.

Reply:
(None of ours do, currently)